### PR TITLE
Refine content and appearance of the project 'Configure' screen 

### PR DESCRIPTION
**Pull request by janfaracik**
This pull request updates the content and minor styling of the project 'Configure' screen to provide descriptions for each section, fix the sidebar for Pipeline projects, and decrease section padding a little. This is part of an ongoing effort to simplify Jenkins' many forms.

**Changes**

* Sections now have descriptions to explain what they're for
* Some section titles have been tweaked, dropping the 'Build' prefix. These are: 'Build Triggers' -> 'Triggers', 'Build Environment' -> 'Environment'. This has been done to help differentiate their labels at a glance.
* Section description margins have been decreased
* Pipeline sections now show in the sidebar

**Before**
image

**After**
image

### Testing done

* Form works as before

### Proposed changelog entries

- Refine content and appearance of the project 'Configure' screen 

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
Please add trailing periods to full sentences. Otherwise it'll be inconsistent or plain weird once a section has more than one sentence

**Comment by janfaracik** [comment]
> Please add trailing periods to full sentences. Otherwise it'll be inconsistent or plain weird once a section has more than one sentence

Done :)

File: war/src/main/scss/components/_section.scss
Comment (daniel-beck) [inline war/src/main/scss/components/_section.scss:38]: Does this still apply, and why?

Comment (janfaracik) [reply war/src/main/scss/components/_section.scss:38]: Yeah, it makes it a little tighter/more balanced than the previous styling from the previous content PR.

**Review by timja — APPROVED** [review]
LGTM - could you file a corresponding PR to pipeline please for consistency?

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
> LGTM - could you file a corresponding PR to pipeline please for consistency?

@janfaracik ^^

**Comment by janfaracik** [comment]
> > LGTM - could you file a corresponding PR to pipeline please for consistency?
> 
> @janfaracik ^^

Heya, I'm on it.

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
Wait, this requires all plugins implementing configuration pages to adapt so things don't look inconsistent? Is this really necessary?

**Pull request merged**
